pkg/api/entries.go

```diff
@@ -301,6 +301,16 @@ func createLogEntry(params entries.CreateLogEntryParams) (models.LogEntry, middl
 	logEntry := models.LogEntry{
 		entryID: logEntryAnon,
 	}
+
+	if api.newEntryPublisher != nil {
+		// Publishing notifications should not block the API response.
+		go func() {
```

Comment:
I don't think the GCP publish() call blocks (per https://pkg.go.dev/cloud.google.com/go/pubsub#Topic.Publish)... should we simply define the interface as "must not block" and defer this down, or leave this as is? Curious to other folks' opinion

Reply:
client.Publish() does not block, but the subsequent [PublishResult.Get()](https://pkg.go.dev/cloud.google.com/go/internal/pubsub#PublishResult.Get) does block until success, error, or timeout. The timeout is 60s, so this call could be blocking up to 60s. The Publish() call does not return an error, so you won't know the status of the call until you add the blocking call.

I prefer to have the Publisher.Publish() interface defined as blocking (must return an error) and leave the option to make it non-blocking up to the caller as is done here.